### ISSUE TITLE
NER Problem

### ISSUE THREAD

**Issue by gabrielluz07** [PKSHATechnology-Research/camphr#51]
Hello!

First of all I would like to thank you for the great work on lib Camphr. It's been very useful to me!
Can you help me with this doubt? I used lib to train a name recognition model (ner) but when I load the model using nlp = (spacy.load ("~ / outputs // 2020-04-30 // 22-28-36 // models // 9 "), and I pass a text (doc = nlp (" I live in Brazil ")), I can't get any entity recognition (doc.ents >> ()). Could you tell me why this is happening?

**Comment by tamuhey** [comment]
Hi, thank you for reporting issue!
First, I want to check that your `nlp` properly contains a ner pipeline.
Please run the below command:

```console
$ ls "~/outputs/2020-04-30/22-28-36/models/9"
```

and run the below python code:

```python
nlp = spacy.load("~/outputs/2020-04-30/22-28-36/models/9")
print(nlp.pipeline)
```

**Comment by gabrielluz07** [comment]
Hello, thanks for the quick response!

The attached images show the commands executed.

![dir](https://user-images.githubusercontent.com/63740244/80856158-cbd69780-8c1d-11ea-9291-92ab676ae156.PNG)
![ner_pipeline](https://user-images.githubusercontent.com/63740244/80856159-ced18800-8c1d-11ea-8331-8c852e951138.PNG)

**Comment by tamuhey** [comment]
Thanks, the ner pipeline is properly loaded.
I think the training isn't done very well, and you should try other hyperparamters to train.
Please see [here](https://camphr.readthedocs.io/en/latest/notes/finetune_transformers.html#advanced-configuration) and configure hyperparamters.

**Comment by gabrielluz07** [comment]
Would you have any suggestions for changing the hyperparamters? I'm still a beginner in machine learning and I'm not sure what changes would work.

**Comment by tamuhey** [comment]
Unfortunately, there is no _correct_ parameters for training.
You should do some trial and error to find good parameters. I hope you can make a good NER model!

**Comment by gabrielosluz** [comment]
Hello.
I apologize for coming back on this issue but I am quite at a loss as to what to do.
I tried to run for 20, 40,100, 200, 500 and 1000 iterations. I also varied the value of the learning rate to 1e-3, 1e-4 and 2e-4. And the model remains unable to recognize any entity.
I tried for data in Portuguese using the neuralmind / bert-base-portuguese-cased model and strangely with 3 phrases for training, the model recognized entities but for larger data sets, like 15 and 50 phrases, it did not.
For the English model, including what is shown in the documentation (bert-base-cased) none of the training variations above work.
So, I was wondering if you have any other tips for getting a model that recognizes entities.

**Comment by tamuhey** [comment]
Hi! was the training loss decreased?

**Comment by gabrielosluz** [comment]
It seems that the loss started low and increased, until it stabilized. The entities I was trying to recognize were part of the training, which makes the inability to recognize even stranger.

05 - "loss":0.8289000988
10 - "loss":1.070409894
25 - "loss":1.3262381554
50 - "loss":1.3723602295
100 - "loss":1.3363217115
150 - "loss":1.3120213747
200 - "loss":1.3691042662
250 - "loss":1.3970358372
300 - "loss":1.3477283716
350 - "loss":1.3162174225
400 - "loss":1.4741054773
450 - "loss":1.4615918398
500 - "loss":1.434394002
550 - "loss":1.4623322487
600 - "loss":1.5252407789
650 - "loss":1.4341518879
700 - "loss":1.4264985323
750 - "loss":1.4794448614
800 - "loss":1.4500057697
850 - "loss":1.5266609192
900 - "loss":1.4918854237
950 - "loss":1.5137566328
999 - "loss":1.4884006977

**Comment by stauntonjr** [comment]
Hello @tamuhey and @gabrielosluz, I didn't check the closed issues when I created mine (https://github.com/PKSHATechnology-Research/camphr/issues/57) which is still open. However looking at this one, I'm curious @gabrielosluz what happens when you type doc.is_nered ? 

My understanding is that, while doc.ents may indeed be empty if the model fails to detects any entities due to insuffcient training or choice of hyperparameters, if the <TrfForNamedEntityRecognition> pipeline component is actually working, then the value of doc.is_nered should be True, regardless of whether any entities were detected. 

![image](https://user-images.githubusercontent.com/36486177/85131941-d668e280-b205-11ea-839c-2875cac0611e.png)


I'd really like to confirm what's going on with this issue.
